docs/serving/queue-extensions.md

```diff
@@ -0,0 +1,40 @@
+# Extending Queue Proxy image with QPoptions
+
+Knative service pods include two containers:
+
+- The user main service container, which is named `user-container`
+- The Queue Proxy - a sidecar named `queue-proxy` that serves as a reverse proxy in front of the `user-container`
+
+Queue Proxy can be extended to offer additional optional features. The QPOptions feature of Queue Proxy allows additional runtime packages to extend Queue Proxy capabilities.
+
+For example, the [security-guard](https://knative.dev/security-guard/README.md) repository provides an extension that uses the QPOptions feature. The [Qpoption](https://knative.dev/security-guard/pkg/qpoption/README.md) package enables users to add additional security features to Queue Proxy.
```

Comment:
Check "QPOptions" vs "QPoptions" here also.

Reply:
will do a full search and align in a separate commit